framework/src/application/node/forger/forger.js

```diff
@@ -220,7 +233,56 @@ class Forger {
 					)} is not a delegate`,
 				);
 			}
+			// Prepare hash-onion
+			const registeredHashOnion = registeredHashOnions[account.address];
```

Comment:
It seems that `registeredHashOnions` are just the seeds. Please rename all occurrences to `registeredOnionSeeds` or `registeredHashOnionSeeds`